cmd/update/volume_move.go

```diff
@@ -259,28 +251,47 @@ A default timeout of 30min is in place but it should be adjusted with the global
 	return nil
 }
 
+// waitForSyncedReplicaOnNode will loop for as long as the context is valid or when a new
+// replica has been fully synced on the target node.
+//
+// in the rare case when a new replica is detected on target node but gets removed before
+// the syncing is completed - most likely due to a background process on control plane that
+// decided to replace/remove said replica - we try to fail this process and avoid the
+// uncessary wait for a state that will not happen.
 func (c *volumeMoveCommand) waitForSyncedReplicaOnNode(ctx context.Context, nsID id.Namespace) error {
+	replicaOnTargetNode := false
+
 	for {
 		select {
 		case <-ctx.Done():
-			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side) - %w", ctx.Err())
+			return fmt.Errorf("timed out waiting for a fully synced replica on [destination] node")
 
 		case <-time.After(time.Second * 8):
 			vol, err := c.getVolume(ctx, nsID)
 			if err != nil {
 				return fmt.Errorf("failed to get updated volume config from api: %s", err)
 			}
 
-			// find the new deployment added with this operation
+			// find the new deployment added with this move operation
+			handledReplica := false
 			for _, rep := range vol.Replicas {
 				if rep.Node.String() == c.toNode {
+					replicaOnTargetNode = true
+					handledReplica = true
+
 					if rep.Promotable {
 						return nil
 					} else if rep.SyncProgress != nil {
-						c.display.ReplicaSyncProgress(ctx, c.writer, *rep.SyncProgress)
+						_ = c.display.ReplicaSyncProgress(ctx, c.writer, *rep.SyncProgress)
+						break
 					}
 				}
 			}
+
+			// no replica on target node now yet we had found one before
+			if !handledReplica && replicaOnTargetNode {
+				return fmt.Errorf("replica on [destination] node removed due to a failure while syncing. please make sure the node has enough capacity.")
+			}
```

Comment:
Should the user then restart the operation?

Reply:
Yes but only after the underlying problem has been taken care of plus I don't want to simply say "restart" due to the possibility that the resolution could imply doing more than just that.
Happy to add it in but I would hate if users would just think "oh let me retry it maybe it was just a fluke" only to see the same error.

Reply:
I think this is probably a bit easier to understand if you mention "fix the problem and then star the operation again". Otherwise, the user may think the operation will continue after they've fixed the problem 
> replica on [destination] node removed due to a failure while syncing. please make sure the node has enough capacity and then restart the application

Reply:
Adding that in